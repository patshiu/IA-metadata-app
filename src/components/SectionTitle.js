import React from 'react';
import PropTypes from 'prop-types';

const SectionTitle = ({ title, count }) => {
  if (count) {
    return (
      <h2>
        {title} ({count})
      </h2>
    );
  } else {
    return <h1>{title}</h1>;
  }
};

SectionTitle.propTypes = {
  title: PropTypes.string.isRequired,
  count: PropTypes.number
};

export default SectionTitle;
