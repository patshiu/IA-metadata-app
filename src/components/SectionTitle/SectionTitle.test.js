import React from 'react';
import renderer from 'react-test-renderer';

import SectionTitle from './SectionTitle';

it('renders section title with item count correct', () => {
  const title = 'Section Title';
  const count = 6;
  const tree = renderer.create(<SectionTitle title={title} count={count} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders section title with no item count correct', () => {
  const title = 'Section Title';
  const count = null;
  const tree = renderer.create(<SectionTitle title={title} count={count} />).toJSON();
  expect(tree).toMatchSnapshot();
});
