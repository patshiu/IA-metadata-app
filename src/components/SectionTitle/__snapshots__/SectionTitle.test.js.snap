// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders section title with item count correct 1`] = `
<h2>
  Section Title
   
  (6)
</h2>
`;

exports[`renders section title with no item count correct 1`] = `
<h2>
  Section Title
   
</h2>
`;
